File: ciphers_algorithms/caesar_cipher/caesar_cipher.py
from string import printable


def encrypt(value: str, step: int) -> str:
    alpha = printable
    result = ""

    for ch in value:
        new_index = (alpha.index(ch) + step) % len(alpha)
        result += alpha[new_index]

    return result


def decrypt(value: str, step: int) -> str:
    step *= -1
    return encrypt(value, step)


def brute_force(value: str):
    alpha = printable
    step = 1
    brute_force_data = {}

    while step <= len(alpha):
        result = decrypt(value, step)
        brute_force_data[step] = result
        step += 1

    return brute_force_data


text = input("Enter Message: ")
step = int(input("Enter Step: "))

encrypt_text = encrypt(text, step)
print(encrypt_text)

decrypt_text = decrypt(encrypt_text, step)
print(decrypt_text)

brute_force_results = brute_force(encrypt_text)
print(brute_force_results)

File: ciphers_algorithms/caesar_cipher/caesar_cipher.cpp
#include <iostream>
#include <string>

using namespace std;


std::string encrypt(std::string value, int step){
    string alpha, result;
    int new_index, pos;

    for (char c{' '}; c <= '~'; ++c)
        alpha += c;

    int length = alpha.size();

    for(auto ch: value){
        pos = alpha.find(ch);
        new_index = (pos + step) % length;

        if(new_index < 0){
            new_index += length;
        }
        result += alpha[new_index];
    }

    return result;
}


std::string decrypt(std::string value, int step){
    step *= -1;
    return encrypt(value, step);
}


int main(){
    std::string message;
    int step;

    std::cout << "Enter Message: ";
    std::getline (std::cin, message);
    std::cout << " " << std::endl;

    std::cout << "Enter step: ";
    std::cin >> step;
    std::cout << " " << std::endl;

    std::string encrypt_result = encrypt(message, step);
    std::cout << "Encrypt Message: " << encrypt_result << std::endl;

    std::string decrypt_result = decrypt(encrypt_result, step);
    std::cout << "Decrypt Message: " << decrypt_result << std::endl;

    return 0;
}